ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Flex, FlexItem, PageSection } from '@patternfly/react-core';
+
+import useURLSearch from 'hooks/useURLSearch';
+import { fetchClustersAsArray } from 'services/ClustersService';
+import { Cluster } from 'types/cluster.proto';
+import ClusterSelect from './ClusterSelect';
+import NamespaceSelect from './NamespaceSelect';
+
+export type Namespace = {
+    metadata: {
+        id: string;
+        name: string;
+    };
+};
+
+function NetworkGraphPage() {
+    const [clusters, setClusters] = React.useState<Cluster[]>([]);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const selectedClusterId = (searchFilter.Cluster as string) || '';
+    const selectedNamespaces = (searchFilter.Namespace as string[]) || [];
+
+    React.useEffect(() => {
```

Comment:
Just to make things cleaner and consistent, we could have hooks to get the clusters and namespaces within this component.

```
const availableClusters = useFetchClusterNames();
const availableNamespaces = useFetchNamespaceNames({ clusterId });
```

and then render the select components like the following:

```
<ClusterSelect selected={selectedClusterId} onSelect={setSelectedClusterId}>
    {availableClusters.map(cluster => {
          return (
              <SelectOption key={cluster.id} value={cluster.id}>{cluster.name}</SelectOption>
          );
    })}
</ClusterSelect>
```
Or something like that. Just throwing out ideas